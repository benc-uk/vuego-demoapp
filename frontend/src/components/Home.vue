<!--
Home page
----------------------------------------------
Ben C, April 2018, Updated for Vue3 2021
-->

<template>
  <div class="p-5 mb-5 dimmed-box rounded-3">
    <div class="container-fluid py-2">
      <h1 class="display-6 fw-bold">Go &amp; Vue.js Demo</h1>
      <p class="col-md-12 fs-5 text-white">
        This is a standard modern web application with a Go RESTful backend and a Vue.js single page application frontend. It has been designed with
        cloud demos &amp; containers in mind. Demonstrating capabilities such as deployment to Azure or Kubernetes, auto scaling, or anytime you want
        something quick and lightweight to run & deploy.
      </p>

      Basic features:
      <ul class="my-4 text-white">
        <li>System status / information view</li>
        <li>Geolocated weather info (from OpenWeather API)</li>
        <li>Realtime monitoring and metrics</li>
        <li>Support for user authentication with Azure AD</li>
      </ul>

      <div class="row justify-content-between">
        <div class="col-md align-self-center">
          <img class="icon mb-2" src="../assets/octocat.png" />
          <a class="btn btn-primary btn-lg" target="_blank" href="https://github.com/benc-uk/vuego-demoapp">GitHub Project</a>
        </div>
        <div class="col-md align-self-center">
          <img class="icon mb-2" src="../assets/container.png" />
          <a class="btn btn-primary btn-lg" target="_blank" href="https://github.com/users/benc-uk/packages/container/package/vuego-demoapp">
            Container Images
          </a>
        </div>
        <div class="col-md align-self-center">
          <img class="icon mb-2" src="../assets/go.svg" />
          <a class="btn btn-primary btn-lg" target="_blank" href="https://azure.microsoft.com/en-gb/develop/go/">Get started</a>
        </div>
      </div>

      <div class="pt-5 h5">
        <a class="btn btn-sm btn-info" href="https://github.com/benc-uk/nodejs-demoapp/wiki/Containerized-Demo-Web-Apps" target="_blank">
          ðŸš€ Demo Apps Collection
        </a>
        &mdash; For more demo apps in different languages &amp; frameworks
      </div>
    </div>
  </div>
</template>

<style scoped>
.dimmed-box {
  background-color: rgba(0, 0, 0, 0.25);
}
.icon {
  height: 60px;
  display: block;
  margin: 0 auto;
}
.col-md {
  text-align: center !important;
}
</style>
