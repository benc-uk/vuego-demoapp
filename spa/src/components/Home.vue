<!--
Home page
----------------------------------------------
Ben C, April 2018
-->

<template>
  <b-jumbotron bg-variant="primary" text-variant="white" border-variant="dark">
    <template #header>
      Go &amp; Vue.js Demo
    </template>
    <template #lead>
      This is a simple web application with a Go server/backend and a Vue.js single page application frontend. It has
      been designed with cloud demos &amp; containers in mind. Demonstrating capabilities such as auto scaling,
      deployment to Azure or Kubernetes, or anytime you want something quick and lightweight to run & deploy.
    </template>
    <hr />

    <div class="dimmed-box">
      <p>
        <img src="../assets/octocat.png" class="icon" />
        <b-button href="https://github.com/benc-uk/vuego-demoapp" variant="info">
          GitHub Project
        </b-button>
        <img src="../assets/docker.png" class="icon" style="margin-left: 50px" />

        <b-button href="https://github.com/users/benc-uk/packages/container/package/vuego-demoapp" variant="info">
          Docker Images
        </b-button>
      </p>

      <hr />

      <p>
        <img src="../assets/go.svg" class="icon" />
        <b-button href="https://azure.microsoft.com/en-gb/develop/go/" variant="info">
          Get started with Azure &amp; Go
        </b-button>
      </p>

      <br />
      <p>Microsoft ‚ù§ Open Source</p>
    </div>
  </b-jumbotron>
</template>

<style scoped>
.dimmed-box {
  background-color: rgba(0, 0, 0, 0.2);
  padding: 1rem;
  border-radius: 0.3rem;
}
.icon {
  width: 60px;
  margin-right: 10px;
}
.jumbotron {
  padding: 2rem;
}
.display-3 {
  padding: 0.1rem;
  font-size: 3rem;
  font-weight: 400;
}
.lead {
  font-size: 1rem;
}
</style>
