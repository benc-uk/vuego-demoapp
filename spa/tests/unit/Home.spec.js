import { createLocalVue, shallowMount } from '@vue/test-utils'
import Home from '@/components/Home.vue'
import flushPromises from 'flush-promises'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('Home.vue', () => {
  it('renders home screen', async () => {
    const wrapper = shallowMount(Home, { localVue, router })

    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })
})
