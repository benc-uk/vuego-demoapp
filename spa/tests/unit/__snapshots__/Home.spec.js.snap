// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders home screen 1`] = `
<b-jumbotron bg-variant="primary" text-variant="white" border-variant="dark">
  <hr>
  <div class="dimmed-box">
    <p><img src="../assets/octocat.png" class="icon">
      <b-button href="https://github.com/benc-uk/vuego-demoapp" variant="info"> GitHub Project </b-button> <img src="../assets/docker.png" class="icon" style="margin-left: 50px;">
      <b-button href="https://github.com/users/benc-uk/packages/container/package/vuego-demoapp" variant="info">
        Docker Images
      </b-button>
    </p>
    <hr>
    <p><img src="../assets/go.svg" class="icon">
      <b-button href="https://azure.microsoft.com/en-gb/develop/go/" variant="info">
        Get started with Azure &amp; Go
      </b-button>
    </p> <br>
    <p>Microsoft ‚ù§ Open Source</p>
  </div>
</b-jumbotron>
`;
